src/views/admin/PersonDetailsView.vue

```diff
@@ -441,10 +561,74 @@
     prepareCreation();
   };
 
+  // When the button "Yes" from the Dialog after filling the form for changing the Klasse is clicked
+  const confirmDialogChangeKlasse = async (): Promise<void> => {
+    changeKlasseConfirmationDialogVisible.value = false;
+    // Find the Orga object for the selected Schule (unchangeable anyways)
+    const organisation: Organisation | undefined = personenkontextStore.workflowStepResponse?.organisations.find(
+      (orga: Organisation) => orga.id === selectedSchule.value,
+    );
+
+    // Find the Orga object for the selected new Klasse
+    const newKlasse: Organisation | undefined = organisationStore.klassen.find(
+      (k: Organisation) => k.id === selectedNewKlasse.value,
+    );
+
+    if (organisation) {
+      // Used to build the Zuordnung of type Schule and keep track of it (Only use it in the template)
+      // This is basically the old Zuordnung in the Schule but since it is already available in ZuordnungenResult we won't be adding this to the finalZuordnungen
+      // It's just better to use this since using an array of Zuordnung (selectedZuordnungen) in the template is not so fun
+      newZuordnung.value = {
+        sskId: organisation.id,
+        rolleId: selectedZuordnungen.value[0]?.rolleId ?? '',
+        klasse: newKlasse?.name,
+        sskDstNr: organisation.kennung ?? '',
+        sskName: organisation.name,
+        rolle:
+          rollen.value?.find((rolle: RolleWithRollenart) => rolle.value === selectedZuordnungen.value[0]?.rolleId)
+            ?.title || '',
+        administriertVon: organisation.administriertVon ?? '',
+        editable: true,
+        typ: OrganisationsTyp.Schule,
+      };
+
+      if (zuordnungenResult.value) {
+        finalZuordnungen.value = zuordnungenResult.value;
+      }
+
+      // Add the new Klasse Zuordnung
+      if (newKlasse) {
+        finalZuordnungen.value.push({
+          sskId: newKlasse.id,
+          rolleId: selectedZuordnungen.value[0]?.rolleId ?? '',
+          sskDstNr: newKlasse.kennung ?? '',
+          sskName: newKlasse.name,
+          rolle:
+            rollen.value?.find((rolle: RolleWithRollenart) => rolle.value === selectedZuordnungen.value[0]?.rolleId)
+              ?.title || '',
+          administriertVon: newKlasse.administriertVon ?? '',
+          editable: true,
+          typ: OrganisationsTyp.Klasse,
+        });
+      }
+    }
+
+    // zuordnungenResult is what we show in the UI and so Zuordnungne of type Klasse shouldn't show up (since they are merged with the ones of type Schule already)
```

Comment:
```suggestion
    // zuordnungenResult is what we show in the UI and so Zuordnungen of type Klasse shouldn't show up (since they are merged with the ones of type Schule already)
```